### ISSUE TITLE
Cannot start beacon node: cannot fetch finalized block root under new state management

### ISSUE THREAD

**Issue by prestonvanloon** [prysmaticlabs/prysm#5462]
```
{"message":"Could not fetch beacon state by root: could not replay blocks for hot state using root: could not process block: could not process block header: parent root 0x9d831e50e7d1c7ae4aced434636e02471f3a34e25fd30b5e6b0bcef8649380cb does not match the latest block header signing root in state 0x0efe7a2bd267da813005721d7821d795571c72ad6565f22a0cc179e9b9d345c1","prefix":"blockchain","severity":"CRITICAL"}
```

I recommend we flip the default to not use new state management as it has not been tested enough.

**Comment by terencechain** [comment]
Working on this issue now

**Comment by terencechain** [comment]
Here's what the complete log looks like:
```
2020/04/16 23:35:45 maxprocs: Leaving GOMAXPROCS=8: CPU quota undefined
{"addr":"http://127.0.0.1:6060/debug/pprof","message":"Starting pprof server","severity":"INFO"}
{"message":"Starting Jaeger exporter endpoint at address = http://jaeger-collector.tracing.svc.cluster.local:14268/api/traces","prefix":"tracing","severity":"INFO"}
{"message":"Using default mainnet config","prefix":"flags","severity":"WARNING"}
{"message":"Starting ETH2 with genesis delay of 0 seconds","prefix":"flags","severity":"WARNING"}
{"message":"Allowing database backups to be triggered from HTTP webhook.","prefix":"flags","severity":"WARNING"}
{"database-path":"/data/beaconchaindata","message":"Checking DB","prefix":"node","severity":"INFO"}
{"message":"ECDSA private key generated","prefix":"p2p","severity":"DEBUG"}
{"message":"Starting beacon node","prefix":"node","severity":"INFO","version":"Prysm/Git commit: cb045dd0e360192fb9e46314b3954f3bf58f4871. Built at: 2020-04-14 21:04:03+00:00"}
{"message":"Starting 8 services: [*p2p.Service *powchain.Service *attestations.Service *blockchain.Service *initialsync.Service *sync.Service *rpc.Service *prometheus.Service]","prefix":"registry","severity":"DEBUG"}
{"message":"Starting service type *p2p.Service","prefix":"registry","severity":"DEBUG"}
{"message":"Starting service type *powchain.Service","prefix":"registry","severity":"DEBUG"}
{"message":"Starting service type *attestations.Service","prefix":"registry","severity":"DEBUG"}
{"message":"Starting service type *blockchain.Service","prefix":"registry","severity":"DEBUG"}
{"message":"Starting service type *initialsync.Service","prefix":"registry","severity":"DEBUG"}
{"message":"Starting service type *sync.Service","prefix":"registry","severity":"DEBUG"}
{"message":"Starting service type *rpc.Service","prefix":"registry","severity":"DEBUG"}
{"message":"Starting service type *prometheus.Service","prefix":"registry","severity":"DEBUG"}
{"address":"0.0.0.0:4000","message":"RPC-API listening on port","prefix":"rpc","severity":"INFO"}
{"message":"You are using an insecure gRPC connection! Provide a certificate and key to connect securely","prefix":"rpc","severity":"WARNING"}
{"address":":9090","message":"Starting prometheus service","prefix":"prometheus","severity":"DEBUG"}
{"message":"Could not fetch beacon state by root: could not replay blocks for hot state using root: could not process block: could not process block header: parent root 0x9d831e50e7d1c7ae4aced434636e02471f3a34e25fd30b5e6b0bcef8649380cb does not match the latest block header signing root in state 0x0efe7a2bd267da813005721d7821d795571c72ad6565f22a0cc179e9b9d345c1","prefix":"blockchain","severity":"CRITICAL"}
```

**Comment by terencechain** [comment]
Able to reproduce the issue via the bad DB.  Got more info:

state slot is `7936` and block slot is `7937`. The issue occurs when processing state to slot `7937`

**Comment by terencechain** [comment]
Resolved via #5466
